File: src/service/ModelBuilder.Models/Data/Connector.cs
using Mb.Models.Enums;
using Mimirorg.TypeLibrary.Enums;
using System;
using System.Collections.Generic;

namespace Mb.Models.Data;

public class Connector
{
    public Guid Id { get; set; }
    public string Name { get; set; }
    public ConnectorDirection Direction { get; set; }
    public string Inside { get; set; }
    public string Outside { get; set; }
    public Guid BlockId { get; set; }
    public List<Attribute> Attributes { get; set; }
    public RelationType TypeConnector { get; set; }
    public Guid TerminalId { get; set; }
    public string Color { get; set; }
}

File: src/client/src/lib/clsses/Connector.ts
import { Attribute, TypeReference } from ".";
import { Direction } from "../enums";

export class Connector {
  id: string;
  name: string;
  direction: Direction;
  aspectObject: string;
  terminalType: string;
  terminalParentType: string;
  inside: string;
  outside: string;
}

export class ConnectorTerminal extends Connector {
  color: string;
  attributes: Attribute[];
  typeReferences: TypeReference[];
}

export class ConnectorRelation extends Connector {}

export class ConnectorPartOf extends ConnectorRelation {}

export class ConnectorFulfilledBy extends ConnectorRelation {}

export class ConnectorHasLocation extends ConnectorRelation {}
